kratos/utilities/sensitivity_builder.cpp

```diff
@@ -0,0 +1,627 @@
+    }
+    else if (KratosComponents<Variable<array_1d<double, 3>>>::Has(rVariable) == true)
+    {
+        auto sensitivity_variables = SensitivityVariables<array_1d<double, 3>>{rVariable};
+        VariableUtils().SetVariable(*sensitivity_variables.pOutputVariable,
+                                    sensitivity_variables.pOutputVariable->Zero(), rNodes);
+    }
+    else
+    {
+        KRATOS_ERROR << "Unsupported variable: " << rVariable << "." << std::endl;
+    }
+    KRATOS_CATCH("");
+}
+
+void SetNodalSolutionStepSensitivityVariablesToZero(const std::vector<std::string>& rVariables,
+                                                    ModelPart::NodesContainerType& rNodes)
+{
+    KRATOS_TRY;
+    for (auto& r_variable : rVariables)
+    {
+        SetNodalSolutionStepSensitivityVariableToZero(r_variable, rNodes);
+    }
+    KRATOS_CATCH("");
+}
+
+template <typename TContainer>
+void SetNonHistoricalSensitivityVariableToZero(std::string const& rVariable, TContainer& rElements)
+{
+    KRATOS_TRY;
+    if (KratosComponents<Variable<double>>::Has(rVariable) == true)
+    {
+        auto sensitivity_variables = SensitivityVariables<double>{rVariable};
+        VariableUtils().SetNonHistoricalVariable(
+            *sensitivity_variables.pOutputVariable,
+            sensitivity_variables.pOutputVariable->Zero(), rElements);
+    }
+    else if (KratosComponents<Variable<array_1d<double, 3>>>::Has(rVariable) == true)
+    {
+        auto sensitivity_variables = SensitivityVariables<array_1d<double, 3>>{rVariable};
+        VariableUtils().SetNonHistoricalVariable(
+            *sensitivity_variables.pOutputVariable,
+            sensitivity_variables.pOutputVariable->Zero(), rElements);
+    }
+    else
+    {
+        KRATOS_ERROR << "Unsupported variable: " << rVariable << "." << std::endl;
+    }
+    KRATOS_CATCH("");
+}
+
+template <typename TContainer>
+void SetNonHistoricalSensitivityVariablesToZero(const std::vector<std::string>& rVariables,
+                                                TContainer& rElements)
+{
+    KRATOS_TRY;
+    for (auto& r_variable : rVariables)
+    {
+        SetNonHistoricalSensitivityVariableToZero(r_variable, rElements);
+    }
+    KRATOS_CATCH("");
+}
+
+void ReplaceDeprecatedNameIfExists(Parameters& rSettings,
+                                   const std::string& rDeprecatedName,
+                                   const std::string& rNewName)
+{
+    KRATOS_TRY;
+    if (rSettings.Has(rDeprecatedName))
+    {
+        KRATOS_WARNING("SensitivityBuilder")
+            << "Setting \"" << rDeprecatedName
+            << "\" is deprecated and will be removed in the future." << std::endl;
+        Parameters value = rSettings[rDeprecatedName].Clone();
+        rSettings.RemoveValue(rDeprecatedName);
+        rSettings.AddValue(rNewName, value);
+    }
+    KRATOS_CATCH("");
+}
+
+std::vector<std::string> ArrayOfStringToVector(Parameters Settings)
```

Comment:
This is quite useful, i use a similar function in #4338. Maybe it can be moved to a more general place?

Reply:
Added it in #4467. We can refactor this in the future if the other PR goes through.